src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -249,7 +249,12 @@ private DocWriteRequest<?> maybeAddExternalVersioning(
   ) {
     if (!config.isDataStream() && !config.shouldIgnoreKey(record.topic())) {
       request.versionType(VersionType.EXTERNAL);
-      request.version(record.kafkaOffset());
+      if (config.hasExternalVersionHeader()) {
+        request.version((Long) record.headers().lastWithName(
+                config.externalVersionHeader()).value());
```

Comment:
When we put this in service, we noticed that even though we used an SMT to write a `long` type header, it didn't come out that way in this code.  We got it to work by using `org.apache.kafka.connect.storage.SimpleHeaderConverter #fromConnectHeader` and then reading it as a String and parsing it to a Long.  Does that sound correct or did we misunderstand something about kafka connect headers?

Reply:
Hey @rjosal-indeed 
Apologies, I missed this before.

Would it be possible to elaborate it? I am not sure if I quite understood how you used SimpleHeaderConverter?

Thanks!

Reply:
yes, the code for this lines with the connect header converter looks like:
```
        final Header versionHeader = record.headers().lastWithName(config.externalVersionHeader());
        final byte[] versionValue = HEADER_CONVERTER.fromConnectHeader(
                record.topic(),
                versionHeader.key(),
                versionHeader.schema(),
                versionHeader.value()
        );
        request.version(Long.parseLong(new String(versionValue)));
```
where HEADER_CONVERTER is an constant `new SimpleHeaderConverter()`

Reply:
Hey @rjosal-indeed 
Great! Thank you so much, this makes sense. 

I think we can go ahead with this change. Also, can you refer to the suggestions mentioned [here](https://github.com/confluentinc/kafka-connect-elasticsearch/pull/697#discussion_r1400510474) (ignore 2nd point) and make the changes accordingly?

Reply:
@sp-gupta ready for review, thanks!

Reply:
Thanks @rjosal-indeed 
There was one import missing due to which the build was failing. I have corrected it. 

Again, Thank you so much for your contribution. Really appreciate!!